File: hw-4B/Python/B-election.py
if __name__ == '__main__':

    """Calculate the election results"""

    dct = {}
    with open('input.txt') as f:
        lines = f.readlines()

    for item in lines:
        name, n = [x for x in item.split()]
        n = int(n)
        if name not in dct:
            dct[name] = n
        else:
            dct[name] += n
    for k, v in sorted(dct.items()):
        print(k, v)

File: hw-4B/C++/B-election.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
#include <algorithm>
#include <fstream>

using namespace std;

int main(){

    // Calculate the election results

    ifstream fin("input.txt");
    int n;
    string name;
    unordered_map <string, int> dct;
    vector  <string> candidates;
    while (fin >> name >> n) {
        if (!dct.count(name)){
            dct[name] = n;
            candidates.push_back(name);
        }
        else{
            dct[name] += n;
        }
    }
    fin.close();
    sort(candidates.begin(), candidates.end());
    for (auto c: candidates){
        cout << c << " "<< dct[c] << "\n";
    }

    return 0;
}